src/main/java/com/android/ex/chips/RecipientEditTextView.java

```diff
@@ -440,21 +439,8 @@ public InputConnection onCreateInputConnection(EditorInfo outAttrs) {
 
     @Override
     public void onSelectionChanged(int start, int end) {
-        // When selection changes, see if it is inside the chips area.
-        // If so, move the cursor back after the chips again.
-        DrawableRecipientChip last = getLastChip();
-        if (last != null && start < getSpannable().getSpanEnd(last)) {
-            // Grab the last chip and set the cursor to after it.
-            setSelection(Math.min(getSpannable().getSpanEnd(last) + 1, getText().length()));
-        }
         super.onSelectionChanged(start, end);
-    }
-
-    @Override
-    public boolean bringPointIntoView(int offset) {
-        boolean handled = mScrollAllowed || mMoreChip != null ? super.bringPointIntoView(offset) : true;
-        mScrollAllowed = false;
-        return handled;
+        setCursorVisible(mSelectedChip == null && getSelectionStart() >= getChipEnd(getLastChip()));
```

Comment:
It's easier to tell if the cursor should be visible when the selection isn't always at the end.